streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -274,6 +289,71 @@ public void setUncaughtExceptionHandler(final Thread.UncaughtExceptionHandler eh
             thread.setUncaughtExceptionHandler(eh);
     }
 
+
+    /**
+     * Find all of the instances of {@link StreamsMetadata} in a {@link KafkaStreams application}
+     *
+     * Note: this is a point in time view and it may change due to partition reassignment.
+     * @return collection containing all instances of {@link StreamsMetadata} in this application
+     */
+    public Collection<StreamsMetadata> allMetadata() {
+        validateIsRunning();
+        return streamsMetadataState.getAllMetadata();
+    }
+
+
+    /**
+     * Find the instances {@link StreamsMetadata} for a given storeName
+     *
+     * Note: this is a point in time view and it may change due to partition reassignment.
+     * @param storeName the storeName to find metadata for
+     * @return  A collection containing instances of {@link StreamsMetadata} that have the provided storeName
+     */
+    public Collection<StreamsMetadata> allMetadataForStore(final String storeName) {
+        validateIsRunning();
+        return streamsMetadataState.getAllMetadataForStore(storeName);
+    }
+
+    /**
+     * Find the {@link StreamsMetadata} for a given storeName and key.
```

Comment:
nit: `Find the {@link StreamsMetadata} instance that contains the given storeName and the corresponding hosted store instance contains the given key.